cylc/flow/suite_files.py

```diff
@@ -826,43 +828,41 @@ def generate_key_store(store_parent_dir, keys_tag):
         if key_file.endswith(UserFiles.Auth.PUBLIC_TAG):
             shutil.move(os.path.join(store_dir, key_file),
                         os.path.join(public_key_loc, '.'))
-            # The public key keeps standard 644 ('-rw-r--r--') file permission
         elif key_file.endswith(UserFiles.Auth.PRIVATE_TAG):
-            loc = shutil.move(os.path.join(store_dir, key_file),
-                              os.path.join(private_key_loc, '.'))
-            # Now lock the private key in its permanent location
-            try:
-                lockdown_private_keys(loc)  # Linux File Permission now 600
-            # Catch any and all errors here, since private keys must be
-            # locked for the security (asymmetric cryptography) to work.
-            except Exception:
-                raise OSError("Unable to lock permissions of private " +
-                              "keys for authentication. Abort.")
+            shutil.move(os.path.join(store_dir, key_file),
+                        os.path.join(private_key_loc, '.'))
 
 
-def ensure_dir_exists(dir_path):
-    """Check for a directory at a path and create it if it does not exist."""
-    if not os.path.exists(dir_path):
-        os.mkdir(dir_path)
+def ensure_keypair_exists(auth_parent_dir, auth_child_dir, tag):
+    """Check if a set of public/private keys exist and if not, create them.
 
+    Args:
+        auth_parent_dir (str):
+            Parent containing public/private key directories (auth_child_dir).
+        auth_parent_dir (str):
```

Comment:
Thanks for adding this
```suggestion
        auth_child_dir (str):
```

Reply:
doh...